### PR TITLE
Support kubernetes

### PR DESCRIPTION
**Pull request by cjellick**
Handle the creation and updating of services created outside of Rancher

### CODE REVIEW

File: resources/content/db/changelog.xml [hunk 1]
```diff
@@ -72,5 +72,6 @@
     <include file="db/core-065.xml"/>
     <include file="db/core-066.xml"/>
     <include file="db/core-067.xml"/>
+    <include file="db/core-101.xml"/>
```
Comment (cjellick) [inline resources/content/db/changelog.xml]: Self reminder: need to fix this before merging

File: code/iaas/agent-instance/src/main/java/io/cattle/platform/agent/instance/dao/AgentInstanceDao.java [hunk 2]
```diff
@@ -30,4 +30,5 @@
 
     void populateNicAndIp(NetworkServiceInfo service);
 
+    List<Long> getAgentProvider(String providedServiceLabel, Long accountId);
```
Comment (alena1108) [inline code/iaas/agent-instance/src/main/java/io/cattle/platform/agent/instance/dao/AgentInstanceDao.java]: Nit: if accountId is never meant to be null, can we change Long to long in the signature?

File: code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalServiceEventManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        if (serviceData != null) {
+            service.putAll(serviceData);
+        }
+
+        try {
+            String create = objectProcessManager.getStandardProcessName(StandardProcess.CREATE, Service.class);
+            String activate = objectProcessManager.getStandardProcessName(StandardProcess.ACTIVATE, Service.class);
+            ProcessUtils.chainInData(service, create, activate);
+            resourceDao.createAndSchedule(Service.class, service);
+        } catch (ProcessCancelException e) {
```
Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalServiceEventManagerImpl.java]: What is the impact of processCancelException (event not being processed)? Do you think we should log it?

**Comment by alena1108** [comment]
LGTM

**Pull request merged**